.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Cargo
+        uses: dtolnay/rust-toolchain@stable
+      
+      - name: Update Cargo
+        run: rustup update stable
+
+      - name: Install Sui CLI
+        run: cargo install --locked --git https://github.com/MystenLabs/sui.git --branch testnet sui --features tracing
```

Comment:
This is going to be very slow. Let me just get the docker build done: https://github.com/zeta-chain/sui-docker/pull/2

I don't think rust has build caching on github actions